frontend/src/component/admin/apiToken/ApiTokenPage/ApiTokenPage.tsx

```diff
@@ -1,19 +1,93 @@
+import { PageHeader } from 'component/common/PageHeader/PageHeader';
+import { CreateApiTokenButton } from 'component/common/ApiTokenTable/CreateApiTokenButton/CreateApiTokenButton';
+import { Search } from 'component/common/Search/Search';
+import { useApiTokenTable } from 'component/common/ApiTokenTable/useApiTokenTable';
 import { useApiTokens } from 'hooks/api/getters/useApiTokens/useApiTokens';
+import { ActionCell } from 'component/common/Table/cells/ActionCell/ActionCell';
+import { CopyApiTokenButton } from 'component/common/ApiTokenTable/CopyApiTokenButton/CopyApiTokenButton';
+import { RemoveApiTokenButton } from 'component/common/ApiTokenTable/RemoveApiTokenButton/RemoveApiTokenButton';
+import useApiTokensApi from 'hooks/api/actions/useApiTokensApi/useApiTokensApi';
 
 export const ApiTokenPage = () => {
     const { hasAccess } = useContext(AccessContext);
-    const { tokens, loading } = useApiTokens();
+    const { tokens, loading, refetch } = useApiTokens();
+    const { deleteToken } = useApiTokensApi();
+
+    const {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        prepareRow,
+        state: { globalFilter },
+        setGlobalFilter,
+        setHiddenColumns,
+        columns,
+    } = useApiTokenTable(tokens, props => (
+        <ActionCell>
+            <CopyApiTokenButton
+                token={props.row.original}
+                permission={READ_API_TOKEN}
+            />
+            <RemoveApiTokenButton
+                token={props.row.original}
+                permission={DELETE_API_TOKEN}
+                remove={async () => {
+                    await deleteToken(props.row.original.secret);
+                    refetch();
+                }}
+            />
+        </ActionCell>
+    ));
 
     return (
         <ConditionallyRender
             condition={hasAccess(READ_API_TOKEN)}
-            show={() => <ApiTokenTable tokens={tokens} loading={loading} />}
+            show={() => (
+                <PageContent
+                    header={
+                        <PageHeader
+                            title={`API access (${rows.length})`}
+                            actions={
+                                <>
+                                    <Search
+                                        initialValue={globalFilter}
+                                        onChange={setGlobalFilter}
+                                    />
+                                    <PageHeader.Divider />
+                                    <CreateApiTokenButton
+                                        permission={CREATE_API_TOKEN}
+                                        path="/admin/api/create-token"
+                                    />
+                                </>
+                            }
+                        />
+                    }
+                >
+                    <ApiTokenTable
+                        loading={loading}
```

Comment:
this is very low-level props API in the ApiTokenTable, maybe the table hook is misplaced. The only reason we do this is because the PageHeader is using rows.length right?

Reply:
Search is placed in the header and needs access to the globalfilter from the react-table useTable hook. Given the constraints and time, this was the best way I could improve the architecture without doing major refactorization.

Reply:
That's what I though. +1 for not changing too much at once